vault/ssorolecredentialsprovider.go

```diff
@@ -94,8 +95,8 @@ func (p *SSORoleCredentialsProvider) getOIDCToken() (token *ssooidc.CreateTokenO
 			return nil, err
 		}
 
-		if p.OIDCTokenCache != nil {
```

Comment:
We need to keep the `if p.OIDCTokenCache != nil {` check in place - it's not present if caching is disabled